tests/integration-service/status-reporting-to-pullrequest.go

```diff
@@ -181,5 +212,95 @@ var _ = framework.IntegrationServiceSuiteDescribe("Status Reporting of Integrati
+				Eventually(func() bool {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, pipelineRun.Name, "", testNamespace)
+					return snapshot == nil
+				}, time.Minute*3, time.Second*5).Should(BeTrue(), fmt.Sprintf("Snashot %s/%s is still be found", snapshot.GetNamespace(), snapshot.GetName()))
+
+				//make build pipelinerun not have snasphot annotation but have invalid image result
+				invalid := "invalid"
+				status := pipeline.PipelineRunStatus{
+					PipelineRunStatusFields: pipeline.PipelineRunStatusFields{
+						Results: []pipeline.PipelineRunResult{
+							{
+								Name:  "IMAGE_DIGEST",
+								Value: *pipeline.NewStructuredValues(invalid),
+							},
+							{
+								Name:  "IMAGE_URL",
+								Value: *pipeline.NewStructuredValues(invalid),
+							},
+							{
+								Name:  "CHAINS-GIT_URL",
+								Value: *pipeline.NewStructuredValues(invalid),
+							},
+							{
+								Name:  "CHAINS-GIT_COMMIT",
+								Value: *pipeline.NewStructuredValues(invalid),
+							},
+						},
+						StartTime: &metav1.Time{Time: time.Now()},
+					},
+					Status: v1.Status{
+						Conditions: v1.Conditions{
+							apis.Condition{
+								Reason: "Completed",
+								Status: "True",
+								Type:   apis.ConditionSucceeded,
+							},
+						},
+					},
+				}
+
+				Eventually(func() bool {
+					pipelineRun, err = f.AsKubeAdmin.HasController.GetComponentPipelineRunWithType(componentName, applicationName, testNamespace, "build", "")
+					pipelineRun, err = f.AsKubeAdmin.TektonController.UpdatePipelineRunStatus(pipelineRun, status)
+					if err != nil {
+						return false
+					}
+					if pipelineRun.Status.Results[0].Value.StringVal != invalid {
+						return false
+					}
+					if err = f.AsKubeAdmin.IntegrationController.RemoveSnapshotAnnotationsFromPipelineRun(pipelineRun); err!= nil {
+						return false
+					}
+
+					if err != nil {
+						return false
+					}
+					if metadata.HasAnnotation(pipelineRun, constants.BuildPipelineRunSnapshotAnnotation) || metadata.HasAnnotation(pipelineRun, constants.SnapshotCreationReportAnnotation) {
+						return false
+					}
+					return true
+				}, time.Minute*3, time.Second*5).Should(BeTrue(), "build piplineRun is not updated as expected")
+
+			})
+
+			It("snapshot creation failure is reported to integration test checkRun", func() {
+				Eventually(func() error {
+					conclusion, text, err := f.AsKubeAdmin.CommonController.Github.GetCheckRunConclusion(integrationTestScenarioPass.Name, componentRepoNameForStatusReporting, prHeadSha, prNumber)
+					if conclusion != constants.CheckrunConclusionFailure || err != nil {
+						GinkgoWriter.Printf("failed to get expected checkRun due to error: %v, actual conclusion is %s: \n", err, conclusion)			
+						return fmt.Errorf("error occurred when checking failing integration test checkRun")
+					}
+					if strings.Contains(*text, "Failed to create snapshot") {
```

Comment:
According to the logic of the tests, you are expecting the snapshot creating fail , 
thus in the second test we ensure that the checkRun contains the fail message , 
in this condition you want to return error if the message of CheckRun is not "Failed ...", right ?
```suggestion
					if !strings.Contains(*text, "Failed to create snapshot") {
```

Reply:
Thanks! Updated.